pbr/config/scene_config.py

```diff
@@ -191,24 +192,61 @@ def configure_scene():
             ]
         }
     )
+
     # Add robot information
+    xbound = (2 * cfg["field"]["border_width"] + cfg["field"]["length"]) / 2
+    ybound = (2 * cfg["field"]["border_width"] + cfg["field"]["width"]) / 2
+
+    # Let the robot's personal space proportional to the field size and inversely proportional to the number of robots by some separation factor
+    # If a maximum recursion depth error is thrown, increase the separation factor here. Be careful to not increase it too much, as this value approaches infinity
+    # the robots will be more likely to be too close / overlap
+    separation_factor = 4
+    r = (
+        (cfg["field"]["width"] * cfg["field"]["length"])
+        - (4 * cfg["field"]["border_width"])
+    ) / (num_robots * separation_factor)
```

Comment:
why not just have an approximate radius for the self robot in config?

Reply:
I was assuming that the changes on the field dimensions would be drastic enough that a constant radius could mess up the randomisation. Should I just put one in a config? Approximate radius for 3 robots with a 54 u^2 field is about 4 and 2/3 units.

Reply:
I'm not sure if I understand what you mean. Should the radius just be how wide the robot model is? It shouldn't be dependent on the field size or how many other robots there are?

Reply:
Its just an arbitrary number that constrains the robot in the field based on the field area and number of robots so when the field config / num robots change the radius adapts too. I.e., the radius is some fraction of the field and num robots in the field. 


Do you want me to put a hard constant instead?

Reply:
Oh so it isn't a measurement in like metres? Is everything in the scene scaled relative to each other? If that's what it is, and it isn't possible to do an actual metres value, then I guess the formula you had is probably fine. But I'd recommend no hardcoding if possible (num_robots and radius_robots).

Reply:
Blender is in metres by default so all of those measurements in the formulae would be in metres too as I just grabbed them from the config.

Where should num robots and radius robots be? I just put them in scene config as thats where I found the num_robots variable before I started this pr

Reply:
Scene config. radius is missing atm in the config